src/firewall/nft.rs

```diff
@@ -0,0 +1,1078 @@
+                                        }
+                                    }
+                                }
+                                _ => continue,
+                            },
+                            _ => continue,
+                        }
+                    }
+                    false
+                };
+
+                for rule in
+                    get_matching_rules_in_chain(&existing_rules, DNATCHAIN, matcher_port_jump)
+                {
+                    batch.delete(schema::NfListObject::Rule(rule));
+                }
+
+                if let Some(v4) = dnat_chain_v4.clone() {
+                    for rule in
+                        get_matching_rules_in_chain(&existing_rules, &v4, match_all_ports_in_range)
+                    {
+                        batch.delete(schema::NfListObject::Rule(rule));
+                    }
+                }
+                if let Some(v6) = dnat_chain_v6.clone() {
+                    for rule in
+                        get_matching_rules_in_chain(&existing_rules, &v6, match_all_ports_in_range)
+                    {
+                        batch.delete(schema::NfListObject::Rule(rule));
+                    }
+                }
+            }
+        }
+
+        if teardown_pf.complete_teardown {
+            let match_dns_dnat = |r: &schema::Rule| -> bool {
+                for statement in &r.expr {
+                    match statement {
+                        // Match any DNS server IP
+                        stmt::Statement::Match(m) => match &m.right {
+                            expr::Expression::String(s) => {
+                                for ip in teardown_pf.config.dns_server_ips {
+                                    if *s == ip.to_string() {
+                                        return true;
+                                    }
+                                }
+                            }
+                            _ => continue,
+                        },
+                        _ => continue,
+                    }
+                }
+                false
+            };
+            for rule in get_matching_rules_in_chain(&existing_rules, DNATCHAIN, match_dns_dnat) {
+                batch.delete(schema::NfListObject::Rule(rule));
+            }
+
+            if let Some(v4) = dnat_chain_v4 {
+                if let Some(c) = get_chain(&existing_rules, &v4) {
+                    batch.delete(schema::NfListObject::Chain(c));
+                }
+            }
+            if let Some(v6) = dnat_chain_v6 {
+                if let Some(c) = get_chain(&existing_rules, &v6) {
+                    batch.delete(schema::NfListObject::Chain(c));
+                }
+            }
+        }
+
+        let rules = batch.to_nftables();
+
+        helper::apply_ruleset(&rules, None, None)?;
+
+        Ok(())
+    }
+}
+
+/// Convert a subnet into a chain name.
+fn get_subnet_chain_name(subnet: IpNet, dnat: bool) -> String {
```

Comment:
I am not a fan of using the subnet as chain name. What is wrong with the existing hash the used in iptables?
We could also just use the network ID I think?
Is there a reason to have one chain per subnet vs chain per network? One per subnet looks incorrect to me, because nft supports both ipv4 and ipv6 at the same time I would expect one chain that matches both the v4 and v6 subnet in there. More chains should also work but doesn't this just make things more complicated?

Reply:
I like using subnets because it reads easier in nftables output (very obvious what each chain does), is completely unambiguous (no chance of a hash collision).

I could definitely combine v4 and v6 subnets into a single chain, but I think separating them is a bit more logical. It doesn't really need more rules, either (we need the same number of rules to actually forward traffic, and separate rules to forward traffic to the chain for v4 and v6 subnets).

Reply:
Yeah I need to dump me a `nft list ruleset` output to see what it actually looks like to better judge what I prefer.

Reply:
So I really do not like the name like that, my main issue is that with the default `podman network create` we juts iterate trough the subnets to assign them, so it is normal to have 10.89.0.0/24, 10.89.1.0/24, 10.89.2.0/24  and so on. This makes it hard for me to follow the chains as only a single digit changes in the name. I have no problem with having the subnet as part of the name but I would prefer if lets say 12 chars of the network ID are added as suffix to make them more distinct to my eyes.

I know we don't really have to look at this often but when we get bug reports we have to and I think a more distinct name will make it easier to find problems (e.g. missing chains/rules)

Reply:
Done. Only 8 characters though (I don't want things to get ridiculously long with IPv6 addresses)